File: apps/payment-service/src/main/java/com/bankusac/payment_service/controller/PaymentController.java
package com.bankusac.payment_service.controller;

import com.bankusac.payment_service.model.Payment;
import com.bankusac.payment_service.service.PaymentService;
import org.springframework.web.bind.annotation.*;

// la ventanilla por donde entran las peticiones HTTP para pagos
@RestController
@RequestMapping("/api/payments")
public class PaymentController {

    private final PaymentService paymentService;

    public PaymentController(PaymentService paymentService) {
        this.paymentService = paymentService;
    }

    // POST /api/payments - procesa un pago con los datos que manda el frontend
    @PostMapping
    public Payment processPayment(@RequestBody ProcessPaymentRequest request) {
        return paymentService.processPayment(request.getTransactionId(), request.getAmount());
    }
}

File: apps/payment-service/src/payment/payment.controller.ts
import {Controller,Get,Param} from '@nestjs/common';import {PaymentService} from './payment.service';
@Controller('api/payments') export class PaymentController{constructor(private readonly s:PaymentService){}@Get()list(){return this.s.list();}@Get('transaction/:id')get(@Param('id')id:string){return this.s.getByTransaction(id);}}
